pkg/controller/directvolumemigrationprogress/trace.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directvolumemigrationprogress
+
+import (
+	"github.com/konveyor/mig-controller/pkg/errorutil"
+	"github.com/konveyor/mig-controller/pkg/settings"
+	migtrace "github.com/konveyor/mig-controller/pkg/tracing"
+	"github.com/opentracing/opentracing-go"
+
+	liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+func (r *ReconcileDirectVolumeMigrationProgress) initTracer(dvmp migapi.DirectVolumeMigrationProgress) (opentracing.Span, error) {
```

Comment:
I don't see where we handle the error from this function. I could just be missing it.

Reply:
The error isn't handled. If this function fails then the reconcile span won't get set and no further child spans will be created under the reconcile span. This uses best-effort to create the span but ignores and continues if an error occurred.